### PR TITLE
chore: remove <-errCh where possible in grpc methods

### PR DESCRIPTION
**Pull request by DmitriyMV**
Simplify code by passing error directly into the pipe closer.

### CODE REVIEW

File: pkg/machinery/client/client.go [hunk 1]
```diff
@@ -1007,16 +997,22 @@ func ReadStream(stream MachineStream) (io.ReadCloser, <-chan error, error) {
 			}
 
 			if data.Metadata != nil && data.Metadata.Error != "" {
+				var err error
+
 				if data.Metadata.Status != nil {
-					errCh <- status.FromProto(data.Metadata.Status).Err()
+					err = status.FromProto(data.Metadata.Status).Err()
```
Comment (smira) [inline pkg/machinery/client/client.go:1003]: I feel we might want to ignore here same errors as above, basically making it `CloseWithError(nil)` is it's Canceled/DeadlineExceeded

**Review by smira — APPROVED** [review]

File: pkg/machinery/client/client.go (hunk 1 above)
Comment (DmitriyMV) [reply pkg/machinery/client/client.go]: Done.

File: pkg/machinery/client/client.go [hunk 2]
```diff
@@ -532,12 +524,12 @@ func (c *Client) DiskUsage(ctx context.Context, req *machineapi.DiskUsageRequest
 }
 
 // Copy implements the proto.MachineServiceClient interface.
```
[the remark this answers is not in the document]
Comment (DmitriyMV) [reply pkg/machinery/client/client.go:529]: 🆗

**Comment by DmitriyMV** [comment]
One cool thing from this, is new error output. For example when you try to get `kubeconfig` from worker node:

Was:
```
$ talosctl -n 172.20.0.5 kubeconfig
rpc error: code = Unimplemented desc = kubeconfig is only available on control plane nodes
error initializing gzip: EOF
```

Now:
$ talosctl -n 172.20.0.5 kubeconfig
error initializing gzip: rpc error: code = Unimplemented desc = kubeconfig is only available on control plane nodes

**Pull request merged**